trade-finance-manager-api/src/graphql/helpers/amendment.helpers.js

```diff
@@ -0,0 +1,189 @@
+const dealTypeCoverStartDate = (facilitySnapshot) => {
+  const { coverStartDate } = facilitySnapshot;
+
+  // if exists - gef
+  if (coverStartDate) {
+    return coverStartDate;
+  }
+
+  let dateConstructed;
+
+  if (facilitySnapshot['requestedCoverStartDate-year'] && facilitySnapshot['requestedCoverStartDate-month'] && facilitySnapshot['requestedCoverStartDate-day']) {
+  // BSS stored as seperate year month day values
+    dateConstructed = new Date(
+      facilitySnapshot['requestedCoverStartDate-year'],
+      facilitySnapshot['requestedCoverStartDate-month'] - 1,
+      facilitySnapshot['requestedCoverStartDate-day'],
+    );
+  } else {
+    dateConstructed = new Date(parseInt(facilitySnapshot.requestedCoverStartDate, 10));
+  }
+
+  return dateConstructed;
+};
+
+// calculates new tenor from amendment coverEndDate and facility coverStartDate.  Requires external api call to calculate tenor
+const calculateAmendmentTenor = async (facilitySnapshot, amendment) => {
+  if (facilitySnapshot && facilitySnapshot?.ukefFacilityType && (facilitySnapshot?.coverStartDate || facilitySnapshot?.requestedCoverStartDate)
+    && amendment && amendment?.coverEndDate) {
+    const { coverEndDate } = amendment;
+    const { ukefFacilityType } = facilitySnapshot;
+
+    // returns coverStartDate from either gef or bss format
+    const coverStartDate = dealTypeCoverStartDate(facilitySnapshot);
+    // formatting for external api call
+    const coverStartDateFormatted = format(new Date(coverStartDate), 'yyyy-MM-dd');
+    const coverEndDateFormatted = format(fromUnixTime(coverEndDate), 'yyyy-MM-dd');
+
+    const updatedTenor = await api.getFacilityExposurePeriod(coverStartDateFormatted, coverEndDateFormatted, ukefFacilityType);
+
+    return updatedTenor.exposurePeriodInMonths;
+  }
+
+  return null;
+};
+
+// checking if amendment completed and if value change accepted fully
+const latestAmendmentValueAccepted = (amendment) => {
+  const { ukefDecision, bankDecision, value, requireUkefApproval } = amendment;
+  const { APPROVED_WITH_CONDITIONS } = DEALS.AMENDMENT_UW_DECISION;
+  const { APPROVED_WITHOUT_CONDITIONS } = DEALS.AMENDMENT_UW_DECISION;
+  const { PROCEED } = DEALS.AMENDMENT_BANK_DECISION;
+
+  const ukefDecisionApproved = ukefDecision?.value === APPROVED_WITH_CONDITIONS || ukefDecision?.value === APPROVED_WITHOUT_CONDITIONS;
+  const bankProceed = bankDecision?.decision === PROCEED;
+
+  if (!requireUkefApproval && value) {
+    return true;
+  }
+
+  if (value && ukefDecisionApproved && bankProceed) {
+    return true;
+  }
+
+  return false;
+};
+
+// checking if amendment completed and if coverEndDate change accepted fully
+const latestAmendmentCoverEndDateAccepted = (amendment) => {
+  const { ukefDecision, bankDecision, coverEndDate, requireUkefApproval } = amendment;
+  const { APPROVED_WITH_CONDITIONS } = DEALS.AMENDMENT_UW_DECISION;
+  const { APPROVED_WITHOUT_CONDITIONS } = DEALS.AMENDMENT_UW_DECISION;
```

Comment:
same as above - this line can be combined with the one above

Reply:
done